src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -4552,31 +4589,101 @@ private BoundExpression BindCollectionLiteralExpression(CollectionCreationExpres
+            {
+                _ = BindValue(syntax.KeyExpression, diagnostics, BindValueKind.RValue);
+                _ = BindValue(syntax.ValueExpression, diagnostics, BindValueKind.RValue);
+                Error(diagnostics, ErrorCode.ERR_CollectionLiteralElementNotImplemented, syntax);
+                return BadExpression(syntax);
+            }
 
-                    case SpreadElementSyntax spreadElementSyntax:
-                        _ = BindValue(spreadElementSyntax.Expression, diagnostics, BindValueKind.RValue);
-                        Error(diagnostics, ErrorCode.ERR_CollectionLiteralElementNotImplemented, syntax);
-                        return BadExpression(syntax);
+            BoundExpression bindSpreadElement(SpreadElementSyntax syntax, BindingDiagnosticBag diagnostics)
+            {
+                var expression = BindRValueWithoutTargetType(syntax.Expression, diagnostics);
+                var builder = new ForEachEnumeratorInfo.Builder();
+                bool hasErrors = !GetEnumeratorInfoAndInferCollectionElementType(syntax, syntax.Expression, ref builder, ref expression, isAsync: false, diagnostics, out _) ||
+                    builder.IsIncomplete;
+                if (hasErrors)
+                {
+                    // PROTOTYPE: Report error.
+                    return BadExpression(syntax);
+                }
 
-                    default:
-                        throw ExceptionUtilities.UnexpectedValue(syntax.Kind());
+                var enumeratorInfo = builder.Build(default);
+                var collectionType = enumeratorInfo.CollectionType;
+                var useSiteInfo = GetNewCompoundUseSiteInfo(diagnostics);
+                var conversion = Conversions.ClassifyConversionFromExpression(expression, collectionType, isChecked: CheckOverflowAtRuntime, ref useSiteInfo);
+                diagnostics.Add(syntax.Expression, useSiteInfo);
+                hasErrors = !conversion.Exists || !conversion.IsImplicit;
+                if (hasErrors)
+                {
+                    GenerateImplicitConversionError(diagnostics, syntax, conversion, expression, collectionType);
+                    // PROTOTYPE:
+                    //// Suppress any additional diagnostics
+                    //diagnostics = BindingDiagnosticBag.Discarded;
+                }
+
+                expression = ConvertForEachCollection(expression, conversion, collectionType, diagnostics);
+                var elementPlaceholder = new BoundValuePlaceholder(syntax.Expression, enumeratorInfo.ElementType);
+                return new BoundCollectionLiteralSpreadElement(
+                    syntax,
+                    expression,
+                    enumeratorInfo,
+                    elementPlaceholder: elementPlaceholder,
+                    elementConversion: null,
+                    addElementPlaceholder: null,
+                    addMethodInvocation: null,
```

Comment:
unclear if these nulls are correct.  if you have `List<long> longs = [1, 2, .. listOfInts]`, shouldn't htere be an element conversion here?  Or maybe i'm not understanding what these mean.

Reply:
The element conversion is generated later, within `BindCollectionInitializerSpreadElementAddMethod()`, when converting the argument for the call to the `List<T>.Add()` method for each element.